### DISCUSSION TITLE
Do you have to re-declare callbacks from imported components.

### DISCUSSION THREAD

**Discussion by samrock5000** [slint-ui/slint#5452] [General] [answered]
i have something like
```slint
import { InitWindow } from "init.slint";
export { InitWindow }

export component MainWindow inherits Window {
   callback realy-cool-function <=> init-window.realy-cool-function;
}

```

I found this to be the only way to access ```realy-cool-function``` from rust.

**Comment by ogoffart** [comment, marked as answer]
If the callback is in a global, you can also access re-exported global directly from native code using the `.global<...>()` function
Otherwise, yes, you need to re-declare callback in the root component.
